homeassistant/components/emoncms/__init__.py

```diff
@@ -21,6 +24,38 @@ async def async_setup_entry(hass: HomeAssistant, entry: EmonCMSConfigEntry) -> b
         entry.data[CONF_API_KEY],
         session=async_get_clientsession(hass),
     )
+    emoncms_unique_id = await emoncms_client.async_get_uuid()
+    if emoncms_unique_id:
```

Comment:
I think this whole block should be in a separate function, maybe even two to make the code more readable.
```suggestion
    _check_unique_id_migration(hass, entry, emoncms_client)
```

```python
def _check_unique_id_migration(hass, entry, emoncms_client) -> None:
    emoncms_unique_id = await emoncms_client.async_get_uuid()
    if emoncms_unique_id:
        if entry.unique_id != emoncms_unique_id:
            _migrate_unique_id(entry, emoncms_unique_id)
    else
        async_create_issue(
            ...
    
```

Reply:
done